storage/pebble/bootstrap.go

```diff
@@ -0,0 +1,138 @@
+		log:                log,
+		db:                 db,
+		leafNodeChan:       make(chan *wal.LeafNode, checkpointLeafNodeBufSize),
+		rootHeight:         rootHeight,
+	}, nil
+}
+
+// batchIndexRegisters
+func (b *RegisterBootstrap) batchIndexRegisters(leafNodes []*wal.LeafNode) error {
+	b.log.Debug().Int("batch_size", len(leafNodes)).Msg("indexing batch of leaf nodes ")
+	batch := b.db.NewBatch()
+	defer batch.Close()
+	for _, register := range leafNodes {
+		payload := register.Payload
+		key, err := payload.Key()
+		if err != nil {
+			return fmt.Errorf("could not get key from register payload: %w", err)
+		}
+
+		registerID, err := keyToRegisterID(key)
+		if err != nil {
+			return fmt.Errorf("could not get register ID from key: %w", err)
+		}
+
+		encoded := newLookupKey(b.rootHeight, registerID).Bytes()
+		err = batch.Set(encoded, payload.Value(), nil)
+		if err != nil {
+			return fmt.Errorf("failed to set key: %w", err)
+		}
+	}
+	err := batch.Commit(pebble.Sync)
+	if err != nil {
+		return fmt.Errorf("failed to commit batch: %w", err)
+	}
+	return nil
+}
+
+// indexCheckpointFileWorker asynchronously indexes register entries in b.checkpointDir
+// with wal.OpenAndReadLeafNodesFromCheckpointV6
+func (b *RegisterBootstrap) indexCheckpointFileWorker() error {
+	b.log.Debug().Msg("started checkpoint index worker")
+	// collect leaf nodes to batch index until the channel is closed
+	batch := make([]*wal.LeafNode, 0, pebbleBootstrapRegisterBatchLen)
+	for leafNode := range b.leafNodeChan {
+		batch = append(batch, leafNode)
+		if len(batch) >= pebbleBootstrapRegisterBatchLen {
+			err := b.batchIndexRegisters(batch)
+			if err != nil {
+				return fmt.Errorf("unable to index registers to pebble in batch: %w", err)
+			}
+			batch = make([]*wal.LeafNode, 0, pebbleBootstrapRegisterBatchLen)
+		}
+	}
+	// index the remaining registers if didn't reach a batch length.
+	err := b.batchIndexRegisters(batch)
+	if err != nil {
+		return fmt.Errorf("unable to index remaining registers to pebble: %w", err)
+	}
+	return nil
+}
+
+// IndexCheckpointFile indexes the checkpoint file in the Dir provided
+func (b *RegisterBootstrap) IndexCheckpointFile(ctx context.Context) error {
+	g, _ := errgroup.WithContext(ctx)
+	b.log.Debug().Msg("indexing checkpoint file for pebble register store")
+	for i := 0; i < pebbleBootstrapWorkerCount; i++ {
+		g.Go(func() error {
+			return b.indexCheckpointFileWorker()
+		})
+	}
+	err := wal.OpenAndReadLeafNodesFromCheckpointV6(b.leafNodeChan, b.checkpointDir, b.checkpointFileName, b.log)
+	if err != nil {
+		return fmt.Errorf("error reading leaf node: %w", err)
+	}
+	if err = g.Wait(); err != nil {
+		return fmt.Errorf("failed to index checkpoint file: %w", err)
+	}
+	b.log.Debug().Msg("checkpoint indexing complete")
+	batch := b.db.NewBatch()
+	defer batch.Close()
+	// update heights atomically to prevent one getting populated without the other
+	err = batch.Set(firstHeightKey(), encodedUint64(b.rootHeight), nil)
```

Comment:
could we make `firstHeightKey` as a value which is initialized by `func init()`. I think I'm commented it before